ConsoleBank/ConsoleBank/Currencies/CurrenciesApi.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Threading.Tasks;
+using Flurl;
+using Flurl.Http;
+using System.Linq;
+
+namespace Currencies
+{
+    public class CurrenciesApi : ICurrenciesApi
+    {
+        private const string CurrencyRatesApiUrl = "https://www.nbrb.by/api/exrates/rates";
+        private const string CurrenciesApiUrl = "https://www.nbrb.by/api/exrates/currencies";
+        
+        public async Task<CurrencyRate> GetCurrenciesRate(string currencyAbbreviation)
+        {
+            var json = CurrencyRatesApiUrl
+                .AppendPathSegment(GetIdFromAbbreviation(currencyAbbreviation, DateTime.Today))
+                .GetJsonAsync<CurrencyRate>();
+
+            return await json;
+        }
+
+        public async Task<CurrencyRate> GetCurrenciesRateOnDate(string currencyAbbreviation, DateTime date)
+        {
+            var json = CurrencyRatesApiUrl
+                .AppendPathSegment(GetIdFromAbbreviation(currencyAbbreviation, date))
+                .SetQueryParams(new
+                {
+                    ondate = date.ToString("yyyy-MM-dd"),
+                }).GetJsonAsync<CurrencyRate>();
+
+            return await json;
+        }
+
+        public async Task<Currency[]> GetCurrencies()
+        {
+            return await CallApi(() => CurrenciesApiUrl.GetJsonAsync<Currency[]>());
+        }
+
+        private int GetIdFromAbbreviation(string currencyAbbreviation, DateTime date)
+        {
+            return GetCurrencies().Result
+                            .Where(x => x.Abbreviation == currencyAbbreviation && x.DateEnd >= date)
+                            .Select(x => x.Id).FirstOrDefault();
```

Comment:
я бы брал single, если мы нашли 2 валюты, то у нас проблема в логике, либо апи поменялось, либо нас баг. Дефолт тоже непонятный кейс, аббревиатуру лучше завалидировать раньше где-то и не нагружать бизнес логику проверкой входных  данных
```suggestion
                            .Select(x => x.Id)
                            .FirstOrDefault();
```